ui/app/utils/all-engines-metadata.ts

```diff
@@ -5,168 +5,203 @@
 
 /**
  * These are all the secret and auth methods, including enterprise.
- * If filtering for enterprise-only engines or a specific category is required, that logic should be done within the consumer of this utility.
  */
 
 export interface EngineDisplayData {
   category?: string;
   displayName: string;
   glyph?: string;
+  mountGroup: string[]; // ['auth'] | ['secret'] | ['auth', 'secret']; // - changed to string[] because of ts error with filterEnginesByMountType - couldn't check mount of type 'never'
   requiresEnterprise?: boolean;
   type: string;
   value?: string;
 }
 
+/**
+ * @param mountType - Given mount type to filter by, e.g., 'auth' or 'secret'.
+ * @param useEnterprise - Optional boolean to indicate if enterprise engines should be included in the results.
```

Comment:
Since this is a boolean, our pattern is to usually prefix with `is` or has` if possible, would you consider renaming to `isEnterprise` for consistency?